### PR TITLE
New Video Renderer

### PR DESCRIPTION
**Pull request by jonathansligh-nimbus**
Added in new video renderer for iOS and Android as well as updating to use the most recent versions of the SDKs.

### CODE REVIEW

File: com.adsbynimbus.nimbus/Editor/IOSPostBuildProcessor.cs [hunk 1]
```diff
@@ -34,7 +35,7 @@ private static void PostProcessBuild_iOS(BuildTarget target, string buildPath)
 				var lines = File.ReadAllLines(path);
 				for(int i = 0 ; i < lines.Length ; i++)
 				{
-					if (lines[i].ToLower().Contains("nimbus"))
+					if (lines[i].ToLower().Contains("nimbussdk"))
```
Comment (JasonSznol) [inline com.adsbynimbus.nimbus/Editor/IOSPostBuildProcessor.cs:38]: The Pod names are case sensitive; it is probably best to replace this with `lines[i].Contains("NimbusSDK")`

File: com.adsbynimbus.nimbus/Editor/NimbusSDKDependencies.xml [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
-      </repositories>-
+      </repositories>
+    </androidPackage>
+    <androidPackage spec="com.adsbynimbus.android:nimbus-vast:2.25.0">
+      <repositories>
+        <repository>https://adsbynimbus-public.s3.amazonaws.com/android/sdks</repository>
+      </repositories>
+    </androidPackage>
+    <!--The below package is needed to resolve gradle issues with the above nimbus-vast package-->
+    <androidPackage spec="io.github.pdvrieze.xmlutil:serialization:0.90.3">
```
Comment (JasonSznol) [inline com.adsbynimbus.nimbus/Editor/NimbusSDKDependencies.xml:14]: Out of curiosity what were the gradle issues? This seems like something that should be updated in the Android SDK

File: com.adsbynimbus.nimbus/Runtime/Scripts/Nimbus.Internal/Network/NimbusClient.cs [hunk 3]
```diff
@@ -39,6 +40,12 @@ public NimbusClient(CancellationTokenSource ctx, NimbusSDKConfiguration configur
 			_ctx = ctx;
 		}
 
+		public void AddHeader(List<KeyValuePair<string, string>> headers) {
```
Comment (JasonSznol) [inline com.adsbynimbus.nimbus/Runtime/Scripts/Nimbus.Internal/Network/NimbusClient.cs:43]: This does not seem to be used anywhere, is there a file missing from this PR?

**Review by JasonSznol — APPROVED** [review]
If the change is NimbusClient is not needed it should be removed or made internal / private; it's best to err on the side of caution when adding things to the public API.

File: com.adsbynimbus.nimbus/Runtime/Scripts/Nimbus.Internal/Network/NimbusClient.cs (hunk 3 above)
Comment (jonathansligh-nimbus) [reply com.adsbynimbus.nimbus/Runtime/Scripts/Nimbus.Internal/Network/NimbusClient.cs:43]: Whoops, test code.

File: com.adsbynimbus.nimbus/Editor/NimbusSDKDependencies.xml (hunk 2 above)
Comment (jonathansligh-nimbus) [reply com.adsbynimbus.nimbus/Editor/NimbusSDKDependencies.xml:14]: `Could not resolve io.github.pdvrieze.xmlutil:serialization:{require [0.86.3, 1.0.0[; prefer 0.86.3}.
     Required by:
         project :launcher > project :unityLibrary > com.adsbynimbus.android:nimbus-vast:2.25.0
      > The consumer was configured to find a library for use during runtime, preferably optimized for Android, as well as attribute 'com.android.build.api.attributes.BuildTypeAttr' with value 'debug', attribute 'com.android.build.api.attributes.AgpVersionAttr' with value '8.3.0'. However we cannot choose between the following variants of io.github.pdvrieze.xmlutil:serialization:0.86.3:
- androidRuntimeElements-published
          - jvmRuntimeElements-published
        All of them match the consumer attributes:
          - Variant 'androidRuntimeElements-published' capability io.github.pdvrieze.xmlutil:serialization:0.86.3 declares a library for use during runtime:
              - Unmatched attributes:
                  - Doesn't say anything about com.android.build.api.attributes.AgpVersionAttr (required '8.3.0')
                  - Doesn't say anything about com.android.build.api.attributes.BuildTypeAttr (required 'debug')
                  - Doesn't say anything about its target Java environment (preferred optimized for Android)
                  - Provides its elements packaged as a jar but the consumer didn't ask for it
                  - Provides release status but the consumer didn't ask for it
                  - Provides attribute 'org.jetbrains.kotlin.platform.type' with value 'androidJvm' but the consumer didn't ask for it`

Comment (jonathansligh-nimbus) [reply com.adsbynimbus.nimbus/Editor/NimbusSDKDependencies.xml:14]: It was a very odd error that required quite a bit of complex gradle changes to even try to fix and none of the fixes @renam and I tried were working.

Comment (renam) [reply com.adsbynimbus.nimbus/Editor/NimbusSDKDependencies.xml:14]: From what we could gather the version of gradle unity uses can't resolve properly the correct version of the xmlutil artifact. There were a few ways we could solve the issue, [xmlutil](https://github.com/pdvrieze/xmlutil?tab=readme-ov-file#notes) readme mentions the issue and has a solution which would require a nimbus-android release. 
The issue does not exists on more recent versions of gradle but we are stuck with what unity supports right now.

**Pull request merged**